### PR TITLE
5 skill08 page

### PR DESCRIPTION
**Pull request by CodesDoWork**
Closes: #5 
Closes: #8

### CODE REVIEW

File: public/locales/de/translation.json [hunk 1]
```diff
@@ -1,38 +1,92 @@
...
+      "requirementsTitle": "Was solltest du an Erfahrungen mitbringen?",
+      "requirements": "Du beherrschst mindestens eine Programmiersprache, z.B. Kotlin, und programmierst Mobile Apps, auch unter Einbeziehung der Verknüpfung zu Webschnittstellen und der Verwendung integrierter Sensoren. Du verstehst englischsprachige Aufgabenstellungen.",
+      "occupation": {
+        "title": "Ein Beruf mit Zukunft und stets am Puls der Zeit",
+        "text": "Die Möglichkeiten der Kommunikation und des Informationsaustauschs nehmen stetig zu. Mit einem Abschluss in der IT bist du gefragt: Im Jahr 2021 wurden ca. 12 % mehr IT-Talente gesucht als im Vorjahr. Der Mobile Applications Developer ist somit einer der am schnellsten wachsenden Berufe.<br />Dein Hauptarbeitsmittel ist dein Laptop. Du bist also in Sachen Arbeitsort und -zeit fast ungebunden, sofern kein Kundentermin ansteht. Das einzige, was du brauchst, ist eine stabile Internetverbindung."
+      },
```
Comment (JonasWanke) [inline public/locales/de/translation.json:70]:
```suggestion
      "short": "Entwickler*innen mobiler Anwendungen nutzen Technologie und Design, um ansprechende und gut funktionierende Anwendungen für Handys zu entwickeln.",
      "description": "Als Softwareentwickler*in mit der Spezialisierung auf Mobile Applications bist du Programmier*in und Designer*in in einem. Du sorgst dafür, dass die Apps auf den mobilen Endgeräten der Benutzer funktionieren, ansprechend aussehen und Spaß machen.<br /><br />Für den Beruf brauchst du ein gewisses Verständnis für Technologie und Design. Du musst die Anforderungen und Wünsche der Auftraggeber*innen verstehen und ein Produkt entwickeln, das zuverlässig, aktualisierbar und für Benutzer*innen leicht zu verstehen und zu verwenden ist.",
      "requirementsTitle": "Was solltest du an Erfahrungen mitbringen?",
      "requirements": "Du beherrschst mindestens eine Programmiersprache, z.&thinsp;B. Dart oder Kotlin, und programmierst mobile Apps, auch unter Einbeziehung der Verknüpfung zu Webschnittstellen und der Verwendung integrierter Sensoren. Du verstehst englischsprachige Aufgabenstellungen.<br /><br />Der Online-Ausscheid besteht aus einem theoretischen und einem praktischen Teil. Für den praktischen Teil kannst du eine Programmiersprache und -umgebung deiner Wahl nutzen. Für die Deutsche Meisterschaft stehen dir dann nur noch das Framework Flutter mit der Programmiersprache Dart sowie native Androidentwicklung mit Kotlin/Java zur Verfügung. Nähere Details folgen in den nächsten Wochen.",
      "occupation": {
        "title": "Ein Beruf mit Zukunft und stets am Puls der Zeit",
        "text": "Die Möglichkeiten der Kommunikation und des Informationsaustauschs nehmen stetig zu. Mit einem Abschluss in der IT bist du gefragt: Im Jahr 2021 wurden ca.&nbsp;12&thinsp;% mehr IT-Talente gesucht als im Vorjahr. Der Mobile Applications Developer ist somit einer der am schnellsten wachsenden Berufe.<br /><br />Dein Hauptarbeitsmittel ist dein Laptop. Du bist also in Sachen Arbeitsort und -zeit fast ungebunden, sofern kein Kundentermin ansteht. Das einzige, was du brauchst, ist eine stabile Internetverbindung."
      },
```

File: public/locales/en/translation.json [hunk 2]
```diff
@@ -1,38 +1,91 @@
...
+      "requirementsTitle": "What experience should you bring with you?",
+      "requirements": "You master at least one programming language, e.g. Kotlin, and program mobile apps, also taking into account the link to web interfaces and the use of integrated sensors. You understand English-language task assignments.",
+      "occupation": {
+        "title": "A Career with a future and always on the cutting edge",
+        "text": "The possibilities of communication and information exchange are constantly increasing. With a degree in IT, you are in demand: in 2021, about 12% more IT talents were sought than in the previous year. The Mobile Applications Developer is therefore one of the fastest growing occupations.<br/>Your main working tool is your laptop, so you are almost unrestricted in terms of work location and time, as long as there is no customer appointment. The only thing you need is a stable internet connection."
+      },
```
Comment (JonasWanke) [inline public/locales/en/translation.json:69]:
```suggestion
      "description": "As a software developer with a specialization in mobile applications, you are a programmer and designer in one. You ensure that the apps work on users' mobile devices, look appealing, and are fun to use.<br /><br />For the job, you need a certain understanding of technology and design. You need to understand the requirements and wishes of the clients and develop a product that is reliable, updatable, and easy for users to understand and use.",
      "requirementsTitle": "What experience should you bring with you?",
      "requirements": "You master at least one programming language, e.g., Dart or Kotlin, and program mobile apps, including the connection to online APIs and the use of integrated sensors. You understand tasks in English.<br /><br />The online competition consists of a theoretical and a practical part. For the practical part, you can use any programming language and environment you like. For the German championships, only the framework Flutter with Dart as a programming language as well as native Android development with Kotlin/Java are available. Further details will be added in the following weeks.",
      "occupation": {
        "title": "A Career with a future and always on the cutting edge",
        "text": "The possibilities of communication and information exchange are constantly increasing. With a degree in IT, you are in demand: in 2021, about 12&thinsp;% more IT talents were sought than in the previous year. The Mobile Applications Developer is therefore one of the fastest growing occupations.<br /><br />Your main working tool is your laptop, so you are almost unrestricted in terms of work location and time, as long as there is no customer appointment. The only thing you need is a stable internet connection."
      },
```

File: src/components/skill/SkillInformation.astro [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+<div class="text-left">
+  <div class="xl:flex items-center">
+    <div class="flex-1 xl:pr-12 xl:border-r border-gray-200">
+      <p class="" set:html={t(`skills.${skillNumber}.description`)} />
+      <br />
+      <br />
+      <Headline>{t(`skills.${skillNumber}.requirementsTitle`)}</Headline>
+      <p>{t(`skills.${skillNumber}.requirements`)}</p>
```
Comment (JonasWanke) [inline src/components/skill/SkillInformation.astro:22]:
```suggestion
      <p set:html={t(`skills.${skillNumber}.requirements`)} />
```

**Review by JonasWanke — CHANGES_REQUESTED** [review]
I really like the design! I've added a bit of text and some events for Skill 08, but can add further content in the following days

File: public/locales/de/translation.json (hunk 1 above)
Comment (CodesDoWork) [reply public/locales/de/translation.json:70]: This text should be on the registration site

File: src/components/EventCard.astro [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  link,
+  hoverTextColor = "black",
+  hoverTextShadowColor = "#ffffff7f",
+} = Astro.props;
+---
+
+<div
+  class={classNames(
+    className,
+    "grid grid-cols-8 grid-rows-11 w-1/2 md:w-3/12 aspect-[18/24]"
```
[the remark this answers is not in the document]
Comment (CodesDoWork) [reply src/components/EventCard.astro:28]: Not my taste

File: src/pages/[lang]/skill08.astro [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+export const getStaticPaths = getStaticLangPaths;
+updateLang(Astro.url.pathname);
+---
+
+<SkillPage
+  skillNumber="08"
+  skillName="Mobile Applications Development"
+  img="https:////live.staticflickr.com/65535/52434467221_676ae1bf70_h.jpg"
+  nationalTeam={[]}
+  timelineEntries={["chancellery2022", "einsteinCup2022", "wsc2022"].map(
```
Comment (JonasWanke) [inline src/pages/[lang]/skill08.astro:15]:
```suggestion
  timelineEntries={["chancellery2022", "wsc2022", "einsteinCup2022"].map(
```

**Review by JonasWanke — CHANGES_REQUESTED** [review]

File: src/components/Dropdown.astro [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        options.map((option) => {
+          const isSelected = option === item;
+
+          return (
+            <li
+              onclick={
+                isSelected
+                  ? undefined
+                  : `location.href = '/${option.toLowerCase()}${pathWithoutLang}'`
+              }
```
Comment (Benjamin-Frost) [inline src/components/Dropdown.astro:43]: This is a very specific action for a component called `Dropdown` we should either allow specifying the `onclick` or rename the component to `LanguageDropdown`

File: src/components/EventCard.astro [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  )}
+>
+  <a
+    target="_blank"
```
Comment (Benjamin-Frost) [inline src/components/EventCard.astro:32]: When using `target="_blank"` you should always set `rel="noopener noreferrer"` (https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever)

File: src/components/EventCard.astro [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    href={link}
+    class="bg-group peer rounded-lg shadow-2xl bg-wsg-orange-500 col-start-1 col-end-8 row-start-1 row-end-11"
+  >
+  </a>
+  <a
+    target="_blank"
```
Comment (Benjamin-Frost) [inline src/components/EventCard.astro:38]: When using `target="_blank"` you should always set `rel="noopener noreferrer"` (https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever)

File: src/components/LinkCard.astro [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  text: string;
+  href: string;
+};
+
+const { name, text, href } = Astro.props;
+---
+
+<a
+  href={href}
+  target="_blank"
```
Comment (Benjamin-Frost) [inline src/components/LinkCard.astro:16]: When using `target="_blank"` you should always set `rel="noopener noreferrer"` (https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever)

File: src/utils/classNames.ts [hunk 10]
```diff
@@ -0,0 +1,2 @@
+export const classNames = (...classes: (false | null | undefined | string)[]): string =>
+    classes.filter(Boolean).join(" ");
```
Comment (Benjamin-Frost) [inline src/utils/classNames.ts:2]: There is a very neat library called `classnames` (https://www.npmjs.com/package/classnames) with lots of cool features that we can use for this. However, considering the little amount of time we have left until deployment, we can refactor this later :)

**Review by Benjamin-Frost — CHANGES_REQUESTED** [review]
I'm amazed by your design skills 🚀  Incredible job!

**Comment by Benjamin-Frost** [comment]
@CodesDoWork Doesn't this PR also close #8? Maybe you forgot to reference it :)

File: src/utils/classNames.ts (hunk 10 above)
Comment (CodesDoWork) [reply src/utils/classNames.ts]: Thank you, didn't know about it

File: src/components/EventCard.astro (hunk 7 above)
Comment (CodesDoWork) [reply src/components/EventCard.astro:32]: Oh yes, thank you! I never heard of it

File: src/components/Dropdown.astro (hunk 6 above)
Comment (CodesDoWork) [reply src/components/Dropdown.astro]: Oh, I forgot...

**Review by JonasWanke — APPROVED** [review]
(I haven't looked at the code in detail)(I haven't looked at the code in detail)

File: astro-i18next.config.mjs [hunk 11]
```diff
@@ -2,14 +2,4 @@
...
-  load: ["client", "server"],
-
-  /** Workaround for for leading '/' in path. Will be fixed in next release.
-   * @see https://github.com/yassinedoghri/astro-i18next/issues/79
-   */
-  i18nextServer: {
-    backend: {
-      loadPath: "./public/locales/{{lng}}/{{ns}}.json",
-    },
-  },
```
Comment (Benjamin-Frost) [inline astro-i18next.config.mjs:14]: The bug below seems to be fixed indeed. I still need the `load: ["client", "server"]` however, to access the translations in react :)

File: src/components/Dropdown.astro [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+      aria-labelledby="listbox-label"
+      aria-activedescendant="listbox-option-3"
+    >
+      {
+        options.map((option) => {
+          const isSelected = option === item;
+
+          return (
+            <li
+              onclick={isSelected ? undefined : onOptionClick(option)}
```
Comment (Benjamin-Frost) [inline src/components/Dropdown.astro:38]: a bit shorter and more readable syntax

```suggestion
              onclick={isSelected && onOptionClick(option)}

```

**Review by Benjamin-Frost — CHANGES_REQUESTED** [review]

File: src/components/Dropdown.astro (hunk 12 above)
Comment (CodesDoWork) [reply src/components/Dropdown.astro:38]: grafik

Comment (Benjamin-Frost) [reply src/components/Dropdown.astro:38]: Nevermind 😆

**Review by Benjamin-Frost — APPROVED** [review]

**Pull request merged**